pkg/component/kubernetes/apiserver/authorization.go

```diff
@@ -0,0 +1,135 @@
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apiserverv1beta1.ConfigSchemeGroupVersion.String(),
+			Kind:       "AuthorizationConfiguration",
+		},
+		Authorizers: []apiserverv1beta1.AuthorizerConfiguration{{Type: "RBAC", Name: "rbac"}},
+	}
+
+	if !k.values.IsWorkerless {
+		authorizationConfiguration.Authorizers = append([]apiserverv1beta1.AuthorizerConfiguration{{Type: "Node", Name: "node"}}, authorizationConfiguration.Authorizers...)
+	}
+
+	for _, webhook := range k.values.AuthorizationWebhooks {
+		config := apiserverv1beta1.AuthorizerConfiguration{
+			Type:    "Webhook",
+			Name:    webhook.Name,
+			Webhook: &webhook.WebhookConfiguration,
+		}
+
+		config.Webhook.ConnectionInfo = apiserverv1beta1.WebhookConnectionInfo{
+			Type:           apiserverv1beta1.AuthorizationWebhookConnectionInfoTypeKubeConfigFile,
+			KubeConfigFile: ptr.To(volumeMountPathStructuredAuthorizationWebhookKubeconfigs + "/" + authorizationWebhookKubeconfigFilename(webhook.Name)),
+		}
+
+		authorizationConfiguration.Authorizers = append(authorizationConfiguration.Authorizers, config)
+	}
+
+	data, err := runtime.Encode(apiServerCodec, authorizationConfiguration)
+	if err != nil {
+		return fmt.Errorf("unable to encode authorization configuration: %w", err)
+	}
+
+	configMap.Data = map[string]string{configMapAuthorizationConfigDataKey: string(data)}
+	utilruntime.Must(kubernetesutils.MakeUnique(configMap))
+	return client.IgnoreAlreadyExists(k.client.Client().Create(ctx, configMap))
+}
+
+func (k *kubeAPIServer) handleAuthorizationSettings(deployment *appsv1.Deployment, configMapAuthorizationConfig *corev1.ConfigMap, secretWebhooksKubeconfigs *corev1.Secret) {
+	if value, ok := k.values.FeatureGates["StructuredAuthorizationConfiguration"]; (!ok || value) &&
+		!versionutils.ConstraintK8sLess130.Check(k.values.Version) {
+		deployment.Spec.Template.Spec.Containers[0].Args = append(deployment.Spec.Template.Spec.Containers[0].Args, fmt.Sprintf("--authorization-config=%s/%s", volumeMountPathStructuredAuthorizationConfig, configMapAuthorizationConfigDataKey))
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts = append(deployment.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+			Name:      volumeNameStructuredAuthorizationConfig,
+			MountPath: volumeMountPathStructuredAuthorizationConfig,
+			ReadOnly:  true,
+		})
+		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: volumeNameStructuredAuthorizationConfig,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{LocalObjectReference: corev1.LocalObjectReference{
+					Name: configMapAuthorizationConfig.Name,
+				}},
+			},
+		})
+	} else {
+		// TODO: Delete this else-branch and everything related to it once we only support a Kubernetes version which
+		//  promotes the StructuredAuthorizationConfiguration feature gate to GA (1.32+ or higher).
```

Comment:
Is the leading space in the second line intentional. It looks a bit strange.

Reply:
Without this, the IDE does not proper color it:

<img width="842" alt="image" src="https://github.com/user-attachments/assets/0db8f586-c0bb-46f4-9449-978ec7f3c93c">
vs.
<img width="842" alt="image" src="https://github.com/user-attachments/assets/a7a96eea-5c9e-49fc-b864-5be991516725">